x/pki/types/message_approve_add_x_509_root_cert.go

```diff
@@ -10,11 +12,13 @@ const TypeMsgApproveAddX509RootCert = "approve_add_x_509_root_cert"
 
 var _ sdk.Msg = &MsgApproveAddX509RootCert{}
 
-func NewMsgApproveAddX509RootCert(signer string, subject string, subjectKeyId string) *MsgApproveAddX509RootCert {
+func NewMsgApproveAddX509RootCert(signer string, subject string, subjectKeyId string, info string) *MsgApproveAddX509RootCert {
 	return &MsgApproveAddX509RootCert{
 		Signer:       signer,
 		Subject:      subject,
 		SubjectKeyId: subjectKeyId,
+		Info:         info,
+		Time:         time.Now().Unix(),
```

Comment:
So, yes, we may have different times on different nodes, and the app hash may be different....
One option is to use consensus time (if we can access it from here). Another options is to have `time` as a required attribute for that constructor to be set by the client. CLI can set it automatically though (using the local time). As it will be set in one place  (user's CLI), it will be the same on all nodes...